### ISSUE TITLE
[BottomNavigation] Bottom Navigation color themer should use greater contrast between colors

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#3115]
### Overview
The MDCBottomNavigationBarColorThemer should apply theme colors so that the background, selected, and unselected items have sufficient contrast. 

For example, the MDC catalog default color scheme uses three colors:

```swift
colorScheme = MDCBasicColorScheme(primaryColor: .init(white: 33 / 255.0, alpha: 1),
                                      primaryLightColor: .init(white: 0.7, alpha: 1),
                                      primaryDarkColor: .init(white: 0, alpha: 1))
```

This produces a BottomNav with a contrast ratio that is too low between the unselected items and the background.

![mdc-bottomnav-catalog-theme](https://user-images.githubusercontent.com/1753199/37614194-76129e2a-2b80-11e8-8508-35f9b2a355df.png)

When creating an MDCBasicColorScheme with a named UIColor like `redColor` or `orangeColor` where the brightness is already maximized, then `primaryLightColor` and `primaryColor` are identical, resulting in disappearing selected items.

![mdc-bottomnav-single-red](https://user-images.githubusercontent.com/1753199/37614227-91912720-2b80-11e8-999f-bade27b3a518.png)
![mdc-bottomnav-single-orange](https://user-images.githubusercontent.com/1753199/37614228-919e1318-2b80-11e8-8d61-e216f17477c3.png)

### Version number
48.0.0

**Comment by romoore** [comment]
Something like

```objectivec
+ (void)applyColorScheme:(id<MDCColorScheme>)colorScheme
    toBottomNavigationBar:(MDCBottomNavigationBar *)bottomNavigationBar {
  bottomNavigationBar.selectedItemTintColor = colorScheme.primaryColor;
}
```

Would look like this:

![simulator screen shot - iphone 4s - 2018-03-19 at 14 30 54](https://user-images.githubusercontent.com/1753199/37614858-2bcb8d0c-2b82-11e8-9918-2d503aaf3813.png)

**Comment by romoore** [comment]
This should be obsolete now that we have semantic color themes available.
